roofit/roofitcore/inc/RooMinimizer.h

```diff
@@ -100,37 +117,103 @@ class RooMinimizer : public TObject {
   void profileStart() ;
   void profileStop() ;
 
-  inline Int_t getNPar() const { return fitterFcn()->NDim() ; }
+  inline Int_t getNPar() const { return fitterFcn()->getNDim() ; }
   inline std::ofstream* logfile() { return fitterFcn()->GetLogFile(); }
   inline Double_t& maxFCN() { return fitterFcn()->GetMaxFCN() ; }
-  
-  const RooMinimizerFcn* fitterFcn() const {  return ( fitter()->GetFCN() ? ((RooMinimizerFcn*) fitter()->GetFCN()) : _fcn ) ; }
-  RooMinimizerFcn* fitterFcn() { return ( fitter()->GetFCN() ? ((RooMinimizerFcn*) fitter()->GetFCN()) : _fcn ) ; }
+
+  const RooAbsMinimizerFcn *fitterFcn() const;
+  RooAbsMinimizerFcn *fitterFcn();
+
+  bool fitFcn() const;
 
 private:
+  template <typename MinimizerFcn = RooMinimizerFcn>
+  RooMinimizer(RooAbsReal &function, MinimizerFcn* /* used only for template deduction */);
```

Comment:
```C++
/* used only for template deduction */
```
Is this really necessary to have a parameter just for template deduction? The type is known at compile time already, so why not just pass the type as the template parameter instead of an object of that time? Eg.
```C++
RooMinimizer<RooMinimizerFcn>(function);
```